src/App.js

```diff
@@ -1,9 +1,116 @@
-function App() {
+import React, { useState, useEffect } from "react";
+import { 
+  AppContainer, Main, TodoContainer, MainText, Line, TodosBody,
+  TodoHeader, TodoList, TodoItem, TodoInput,
+  Input, Button, RemoveButton 
+} from './styles';
+import { GlobalStyle } from './styles';
+
+const App = () => {
+  const [todos, setTodos] = useState([]); // 할 일 목록 관리
+  const [newTodo, setNewTodo] = useState(""); // 새로 추가될 할 일 관리
+  const [currentDate, setCurrentDate] = useState(""); // 현재 날짜 저장
+
+  // 컴포넌트 마운트 시 현재 날짜 설정
+  useEffect(() => {
+    const today = new Date();
+    const days = ["일", "월", "화", "수", "목", "금", "토"];
+    setCurrentDate(
+      `${today.getMonth() + 1}월 ${today.getDate()}일 ${days[today.getDay()]}요일`
+    );
+  }, []);
+
+  // 할 일 추가
+  const addTodo = () => {
+    if (newTodo.trim()) {
+      // 새로운 투두 객체를 생성 (complete 속성 추가)
+      setTodos([...todos, { text: newTodo, complete: false }]);
+      setNewTodo(""); // 입력창 초기화
+    }
+  };
+
+  // 할 일 삭제
+  const removeTodo = (index) => {
+    const updatedTodos = todos.filter((_, i) => i !== index);
+    setTodos(updatedTodos);
+  };
```

Comment:
`useCallback` 함수로 감싸서 해당 App.js가 렌더링 될 때마다 불필요하게 리렌더링 되는걸 막으면 최적화에 더 좋을 것 같습니당

```suggestion
  // 할 일 삭제
  const removeTodo = useCallback((index) => {
    const updatedTodos = todos.filter((_, i) => i !== index);
    setTodos(updatedTodos);
  }, []);
```